IQSS/dataverse
# fix: issues in exporters and citations for PermaLink/non-DOI PIDs

**What this PR does / why we need it**:

This PR fixes three issues with PermaLink PIDs in metadata exporters and citations.

1. DDI & DDI HTML exporters: Previously, the export included `<IDNo agency="perma">perma:N4H-/M015D56B</IDNo>` (DDI XML) and "Identification Number: perma:N4H-/M015D56B" (DDI HTML). This change removes the extra slash in the middle of the identifier.

2. EndNote XML citation: Previously, the citation included `<electronic-resource-num>perma/N4H-/M015D56B</electronic-resource-num>`. With this change, `perma/` is replaced with `perma:` and the extra slash in the middle of the identifier is also removed. (Note, this is the only change that also affects DOIs: for DOIs, `doi/` is replaced with `doi:`.)

3. BibTeX citation: Previously, `doi = {...}` was always included in the citation. With this change, it is only included when the PID is actually a DOI. A `url = {...}` is always included (this was already the case).

**Which issue(s) this PR closes**:

In combination with #10775 and #10759 (or #10615), this addresses #10769 and #10768

- Closes #10769
- Closes #10768
- Closes #11165

**Special notes for your reviewer**:

\-

**Suggestions on how to test this**:
(from @qqmyers)
1) Configure permalink PID providers w/  "" and "/" separators, create datasets using them and verify that the DDI/DDI Html exports do not/do contain the / separator (respectively). Confirm that  a Fake DOI provider (or test DataCite provider) still work as before w.r.t. how they appear in the DDI/DDI Html exports.

2) Verify that the EndNote citation for datasets using DOIs and PermaLinks include the PIDs as specified in the best option as discussed in [this comment](https://github.com/IQSS/dataverse/pull/10790#issuecomment-2433406188) and below. Handles could also be tested but I'm not sure if there's a test setup for that.

3) Verify that the BibTeX citation format uses `doi=` for DOIs and only `url=` for PermaLinks/Handles (if we can test them). (pre-PR PermaLinks would have an extra `doi=` line.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

\-

**Is there a release notes update needed for this change?**:
Provided


**Additional documentation**:

\-
